tests/models/vits/test_modeling_vits.py

```diff
@@ -177,6 +179,30 @@ def test_model_forward(self):
         config_and_inputs = self.model_tester.prepare_config_and_inputs()
         self.model_tester.create_and_check_model_forward(*config_and_inputs)
 
+    @require_torch_multi_gpu
+    # override to make test deterministic across GPUs
+    def test_multi_gpu_data_parallel_forward(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        config.use_stochastic_duration_prediction = False
+
+        # move input tensors to cuda:O
+        for key, value in inputs_dict.items():
+            if torch.is_tensor(value):
+                # make all elements of the batch the same -> ensures the output seq lengths are the same for DP
```

Comment:
make all elements of the batch the same **seq length**

Reply:
maybe mention the output length is determined by the first entry of each element in the batch.

Reply:
TWe have to make each of the elements in the batch strictly the **same** -> the output length is a function of the inputs, not just the input lengths. If the inputs are not the same, we get different output dims, so can't do DP

Reply:
ok, thanks for explaning!